addons/mail/static/src/core/common/composer.js

```diff
@@ -219,13 +223,27 @@ export class Composer extends Component {
         );
         onMounted(() => {
             this.ref.el?.scrollTo({ top: 0, behavior: "instant" });
-            if (!this.props.composer.text) {
+            if (!this.props.composer.composerText) {
                 this.restoreContent();
             }
         });
         onWillUnmount(() => {
             this.props.composer.isFocused = false;
         });
+        const composerProxy = reactive(this.props.composer, () => {
+            if (this.status === 2 /* DESTROYED */) {
+                return;
+            }
+            const composerHtml = composerProxy.composerHtml;
+            if (this.updateFromEditor) {
+                return;
+            }
+            if (!this.editor?.editable) {
+                return;
+            }
+            setElementContent(this.editor?.editable, composerHtml);
+        });
+        void composerProxy.composerHtml; // start observing
```

Comment:
Ideally, we need `this.editor.shared.history.addStep()`

Reply:
What are steps for? I don't think we care about adding a step when we are just synchronizing the value, it's not user input and we are not supposed to go backwards from there. If anything, we should clear the full history.